hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -57,6 +58,7 @@
   public static final String HOODIE_TABLE_TYPE_PROP_NAME = "hoodie.table.type";
   public static final String HOODIE_TABLE_VERSION_PROP_NAME = "hoodie.table.version";
   public static final String HOODIE_TABLE_PRECOMBINE_FIELD = "hoodie.table.precombine.field";
+  public static final String HOODIE_TABLE_PARTITION_COLUMNS = "hoodie.table.partition.columns";
```

Comment:
I think we should persist the key generator class. and not the partition columns themselves? let me think over this more.

Reply:
Hi @vinothchandar , we need the partition schema for partition prune for spark sql. So the partition columns is need for that. Or we cannot get the partition schema.

Reply:
how does this work for existing tables? do we need an upgrade-downgrade step for writing this to `hoodie.properties`?

Reply:
For existing tables without store the partition columns, we query it as a non-partitioned table.